File: archive-application/src/main/java/site/archive/dto/v2/MyArchiveResponseDto.java
package site.archive.dto.v2;

import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import site.archive.common.DateTimeUtils;
import site.archive.domain.archive.Archive;
import site.archive.domain.archive.CoverImageType;
import site.archive.domain.archive.Emotion;

@RequiredArgsConstructor
@Builder
@Getter
public class MyArchiveResponseDto {

    private final Long archiveId;
    private final String name;
    private final String watchedOn;
    private final Emotion emotion;
    private final String mainImage;
    private final Boolean isPublic;
    private final CoverImageType coverImageType;
    private final Long authorId;
    private final Long likeCount;
    private final Long dateMilli;

    public static MyArchiveResponseDto from(Archive archive, long dateMilli) {
        var author = archive.getAuthor();
        var likeCount = archive.getLikes().stream()
                               .filter(like -> !like.getIsDeleted()).count();
        return MyArchiveResponseDto.builder()
                                   .archiveId(archive.getId())
                                   .name(archive.getName())
                                   .watchedOn(archive.getWatchedOn().format(DateTimeUtils.getYymmddFormatter()))
                                   .emotion(archive.getEmotion())
                                   .mainImage(archive.getMainImage())
                                   .isPublic(archive.getIsPublic())
                                   .coverImageType(archive.getCoverImageType())
                                   .authorId(author.getId())
                                   .likeCount(likeCount)
                                   .dateMilli(dateMilli)
                                   .build();
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v2/MyArchiveResponseDto.kt
package site.archive.dto.v2

import site.archive.common.yymmddFormatter
import site.archive.domain.archive.Archive
import site.archive.domain.archive.CoverImageType
import site.archive.domain.archive.Emotion

class MyArchiveResponseDto(
    val archiveId: Long,
    val name: String,
    val watchedOn: String,
    val emotion: Emotion,
    val mainImage: String,
    val isPublic: Boolean,
    val coverImageType: CoverImageType,
    val authorId: Long,
    val likeCount: Long,
    val dateMilli: Long
) {
    companion object {
        @JvmStatic
        fun from(archive: Archive, dateMilli: Long): MyArchiveResponseDto {
            val author = archive.author
            val likeCount = archive.likes.stream()
                .filter { !it.isDeleted }
                .count()
            return MyArchiveResponseDto(
                archiveId = archive.id,
                name = archive.name,
                watchedOn = archive.watchedOn.format(yymmddFormatter),
                emotion = archive.emotion,
                mainImage = archive.mainImage,
                isPublic = archive.isPublic,
                coverImageType = archive.coverImageType,
                authorId = author.id,
                likeCount = likeCount,
                dateMilli = dateMilli
            )
        }
    }
}